lib/beaker/host.rb

```diff
@@ -224,10 +224,26 @@ def get_ip
       @logger.warn("Uh oh, this should be handled by sub-classes but hasn't been")
     end
 
+    # Determine the ip address using logic specific to the hypervisor
+    def get_public_ip
+      case host_hash[:hypervisor]
+      when 'ec2'
+        if host_hash[:instance]
+          host_hash[:instance].ip_address
+        else
+          # In the case of using ec2 instances with the --no-provision flag, the ec2
+          # instance object does not exist and we should just use the curl endpoint
+          # specified here:
+          # http://docs.aws.amazon.com/AWSEC2/latest/UserGuide/using-instance-addressing.html
+          execute("curl http://169.254.169.254/latest/meta-data/public-ipv4").strip
```

Comment:
Not from what I can tell @kevpl, but I just realized that amazon documents using `wget` for windows boxes; I should add that in now too.